firebase-perf/src/main/java/com/google/firebase/perf/metrics/AppStartTrace.java

```diff
@@ -337,42 +346,50 @@ public synchronized void onActivityResumed(Activity activity) {
     appStartActivity = new WeakReference<Activity>(activity);
 
     onResumeTime = clock.getTime();
-    this.appStartTime = firebaseStartupTime;
     this.startSession = SessionManager.getInstance().perfSession();
     AndroidLogger.getInstance()
         .debug(
             "onResume(): "
                 + activity.getClass().getName()
                 + ": "
-                + this.appStartTime.getDurationMicros(onResumeTime)
+                + getClassLoadTime().getDurationMicros(onResumeTime)
                 + " microseconds");
 
     // Log the app start trace in a non-main thread.
     executorService.execute(this::logAppStartTrace);
 
-    if (!isExperimentTTIDEnabled && isRegisteredForLifecycleCallbacks) {
+    if (!isExperimentTTIDEnabled) {
       // After AppStart trace is logged, we can unregister this callback.
       unregisterActivityLifecycleCallbacks();
     }
   }
 
-  private void logExperimentTtid(TraceMetric.Builder metric) {
-    transportManager.log(metric.build(), ApplicationProcessState.FOREGROUND_BACKGROUND);
+  /** Helper for logging all experiments in one trace. */
+  private void logExperimentTrace(TraceMetric.Builder metric) {
+    if (this.preDrawPostTime == null
+        || this.preDrawPostAtFrontOfQueueTime == null
+        || this.onDrawPostAtFrontOfQueueTime == null) {
+      return;
```

Comment:
I'm worried if we would drop traces because of these checks? Are we sure every app start trace would contains these timers? I would rather recommend capturing them and explore in the analysis to see how common is it to have these values as NULL.

Reply:
That's due to limitation that we only want to have 1 experiment trace, and add the rest as subtraces. If I send each trace by themselves then we would have multiple traces, which I don't have a problem with but it's just that it doesn't match our ariane so I wonder if it's okay.